### PR TITLE
[ICDS Dashboard] State level users are able to view National level data by clicking 'National' from the breadcrumb issue fix

### PR DESCRIPTION
**Pull request by challabeehyv**
https://dimagi-dev.atlassian.net/browse/ICDS-1303

##### SUMMARY
State level users are able to view National level data by clicking 'National' from the breadcrumb

##### FEATURE FLAG
No

### CODE REVIEW

**Review by czue — APPROVED** [review]

File: custom/icds_reports/templates/icds_reports/icds_app/program-summary.directive.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
                 <span class="breadcrump-element" ng-bind="$ctrl.label"></span>
             </div>
             <div class="row">
-                <span class="location-breadcump" ng-click="$ctrl.getDisableIndex() < 0 && $ctrl.moveToLocation('national', -1)">National </span>
+                <span class="location-breadcump" ng-click="$ctrl.haveAccessToAllLocations && $ctrl.moveToLocation('national', -1)">National </span>
```
Comment (calellowitz) [inline custom/icds_reports/templates/icds_reports/icds_app/program-summary.directive.html]: `ng-click` is the function that gets fired when you click on the breadcrumb. is the idea here that if the first part returns `false` js will short-circuit and not evaluate the second part?

**Review by calellowitz — CHANGES_REQUESTED** [review]
`getDisableIndex` is used in a lot of places in the dashboard. this looks like it fixes some of them but that the underlying problem still exists on a lot of pages. It also seems to be used elsewhere for location permission checking beyond just blocking national. If it's behavior is wrong, I think we need to also fix that function, rather than just replace it in a few cases.

File: custom/icds_reports/templates/icds_reports/icds_app/program-summary.directive.html (hunk 1 above)
Comment (challabeehyv) [reply custom/icds_reports/templates/icds_reports/icds_app/program-summary.directive.html]: yes If we does not have access to it

**Comment by calellowitz** [comment]
This was superseded by https://github.com/dimagi/commcare-hq/pull/26848

**Pull request closed without merging**